aspnetcore/host-and-deploy/scaling-aspnet-apps/scaling-aspnet-apps.md

```diff
@@ -0,0 +1,320 @@
+ms.author: alexwolf
+ms.custom: mvc
+ms.date: 8/31/2022
+---
+# Deploying and scaling an ASP.NET Core app on Azure Container Apps
+
+Apps deployed to Azure that experience intermittent high demand benefit from scalability to meet demand. Scalable apps can scale out to ensure capacity during workload peaks and return to normal automatically when the peak drops. Horizontal scaling (scaling out) adds new instances of a resource, such as VMs or database replicas. This article demonstrates how to deploy a horizontally scalable ASP.NET Core app to [Azure container apps](/azure/container-apps/overview) by completing the following tasks:
+
+1. [Setup the sample project](#set-up-the-sample-project)
+1. [Deploy the app to Azure Container Apps](#deploy-the-app-to-azure-container-apps)
+1. [Scale and troubleshoot the app](#scale-and-troubleshoot-the-app)
+1. [Create the Azure Services](#create-the-azure-services)
+1. [Connect the Azure services](#connect-the-azure-services)
+1. [Configure and redeploy the app](#configure-and-redeploy-the-app)
+
+This article uses Razor Pages, but most of it applies to other ASP.NET Core apps.
+
+In some cases, basic ASP.NET Core apps are able to scale without special considerations. However, apps that utilize certain framework features or architectural patterns require extra configurations, including the following:
+
+* **Secure form submissions**: Razor Pages, MVC and Web API apps often rely on form submissions. By default these apps use [cross site forgery tokens](xref:security/anti-request-forgery) and internal data protection services to secure requests. When deployed to the cloud, these apps must be configured to manage data protection service concerns in a secure, centralized location.
+
+* **SignalR circuits**: Blazor Server apps require the use of a centralized [Azure SignalR service](/azure/azure-signalr/signalr-overview) in order to securely scale. These services also utilize the data protection services mentioned previously.
+
+* **Centralized caching or state management services**: Scalable apps may use [Azure Cache for Redis](/azure/azure-cache-for-redis/cache-overview) to provide distributed caching. [Azure storage](/azure/storage/common/storage-introduction) may be needed to store state for frameworks such as [Microsoft Orleans](/dotnet/orleans/overview), which can help write apps that manage state across many different app instances.
+
+> [!NOTE]
+> The steps in this article demonstrate how to properly address these concerns by deploying a scalable app to Azure Container Apps. Most of the concepts in this tutorial also apply when scaling [Azure App Service](/azure/app-service/overview) instances.
+
+## Set up the sample project
+
+You can use the GitHub Explorer sample app to follow along with this tutorial. Clone the app from GitHub using the following command:
+
+```dotnetcli
+git clone "https://github.com/dotnet/AspNetCore.Docs.Samples.git"
```

Comment:
This works OK because the [AspNetCore.Docs.Samples](https://github.com/dotnet/AspNetCore.Docs.Samples) is small. @tdykstra  or @guardrex  might know where/how to create a repo for just this sample to make it even simpler.

Reply:
> @tdykstra or @guardrex might know where/how to create a repo for just this sample to make it even simpler.

Sorry, I don't know where to create a single-sample repo either.